modal/_utils/hash_utils.py

```diff
@@ -12,6 +12,7 @@ def _update(hashers, data: Union[bytes, IO[bytes]]):
         for hasher in hashers:
             hasher.update(data)
     else:
+        assert isinstance(data, IO)
```

Comment:
Odd that this is necessary!

Reply:
It turns out to be https://github.com/microsoft/pyright/issues/5697

Also this broke our CI once it landed which is weird because it looks like it passed on the branch 🤔

Reply:
@mwaskom My bad for delay, had been deep in interview-land 😅

Reply:
No worries, I got to learn some interesting stuff about typing edge cases :D